src/Spark.Engine/Service/FhirServiceExtensions/PatchApplicationService.cs

```diff
@@ -0,0 +1,220 @@
+            var variable = Expression.Variable(propertyInfo.PropertyType, "item");
+            var block = Expression.Block(
+                new[] { variable },
+                Expression.Assign(
+                    variable,
+                    Expression.MakeIndex(result, propertyInfo, new[] { Expression.Constant(source) })),
+                Expression.Call(result, GetMethod(result.Type, "RemoveAt"), Expression.Constant(source)),
+                Expression.Call(
+                    result,
+                    GetMethod(result.Type, "Insert"),
+                    Expression.Constant(Math.Max(0, destination - 1)),
+                    variable));
+            return block;
+        }
+
+        private static Expression InsertValue(Expression result, ConstantExpression valueExpression)
+        {
+            return result switch
+            {
+                IndexExpression indexExpression => Expression.Call(
+                    indexExpression.Object,
+                    GetMethod(indexExpression.Object!.Type, "Insert"),
+                    new[] { indexExpression.Arguments[0], valueExpression }),
+                _ => result
+            };
+        }
+
+        private static Expression AddValue(Expression result, ConstantExpression value)
+        {
+            return result switch
+            {
+                MemberExpression me when me.Type.IsGenericType
+                                         && GetMethod(me.Type, "Add") != null =>
+                    Expression.Block(
+                        Expression.IfThen(
+                            Expression.Equal(me, Expression.Default(result.Type)),
+                            Expression.Throw(Expression.New(typeof(InvalidOperationException)))),
+                        Expression.Call(me, GetMethod(me.Type, "Add"), value)),
+                MemberExpression me => Expression.Block(
+                    Expression.IfThen(
+                        Expression.NotEqual(me, Expression.Default(result.Type)),
+                        Expression.Throw(Expression.New(typeof(InvalidOperationException)))),
+                    Expression.Assign(me, value)),
+                _ => result
+            };
+        }
+
+        private static Expression DeleteValue(Expression result)
+        {
+            return result switch
+            {
+                IndexExpression indexExpression => Expression.Call(
+                    indexExpression.Object,
+                    GetMethod(indexExpression.Object!.Type, "RemoveAt"),
+                    indexExpression.Arguments),
+                MemberExpression me when me.Type.IsGenericType
+                                         && typeof(List<>).IsAssignableFrom(me.Type.GetGenericTypeDefinition()) =>
+                   Expression.Call(me, GetMethod(me.Type, "Clear")),
+                MemberExpression me => Expression.Assign(me, Expression.Default(me.Type)),
+                _ => result
+            };
+        }
```

Comment:
Can we move this methods into internal classes, like AddPatchOperation, DeletePatchOperation and so on. And let them all inherit from an internal interface called IPathOperation with a single method Execute.

Reply:
I don't agree that the command pattern is warranted here. Each operation is pure and I don't see the need to instantiate it one use. If I'm right, then the added benefit that you would get would be to create separate tests for each command. However I think this is just as well achieved by having targeted tests.

Obviously open to discuss this if there is a concern I didn't get.

Reply:
Let's leave this as it is for now